packages/core/test/tracing/index_test.js

```diff
@@ -7,49 +7,77 @@
 const expect = require('chai').expect;
 const sinon = require('sinon');
 const semver = require('semver');
-const tracing = require('../../src/tracing');
+const proxyquire = require('proxyquire');
+
 const normalizeConfig = require('../../src/util/normalizeConfig');
 const kafkaJs = require('../../src/tracing/instrumentation/messaging/kafkaJs');
+const rdKafka = require('../../src/tracing/instrumentation/messaging/rdkafka');
 const grpc = require('../../src/tracing/instrumentation/protocols/grpc');
 const grpcJs = require('../../src/tracing/instrumentation/protocols/grpcJs');
 
 const mochaSuiteFn = semver.satisfies(process.versions.node, '>=8.13.0') ? describe : describe.skip;
 
 mochaSuiteFn('[UNIT] tracing/index', function () {
+  let tracing;
+
   let activateStubGrpc;
   let activateStubGrpcJs;
   let activateStubKafkaJs;
+  let setKafkaTracingConfigStubKafkaJs;
+  let setKafkaTracingConfigStubRdKafka;
 
   before(() => {
     activateStubGrpc = sinon.stub(grpc, 'activate');
     activateStubGrpcJs = sinon.stub(grpcJs, 'activate');
     activateStubKafkaJs = sinon.stub(kafkaJs, 'activate');
+    setKafkaTracingConfigStubKafkaJs = sinon.stub(kafkaJs, 'setKafkaTracingConfig');
+    setKafkaTracingConfigStubRdKafka = sinon.stub(rdKafka, 'setKafkaTracingConfig');
   });
 
-  it('deactivate instrumentation via config', () => {
-    const config = { tracing: { disabledTracers: ['grpc', 'kafka'] } };
-    normalizeConfig(config);
+  beforeEach(() => {
+    // requiring tracing/index via proxyquire gives us a module in pristine state everytime, in particular with the
+    // tracingActivated flag reset.
+    tracing = proxyquire('../../src/tracing', {});
```

Comment:
There were some issues with this test suite due to the fact that we re-used the same (stateful) instance of `require('../../src/tracing')` for all tests. In particular, the `activate` method was not executed, except for the first test in the suite.

Reply:
Maybe this is cased by [this line](https://github.com/instana/nodejs/pull/527/files#diff-e2d7b3fbf936c12cd8fa74f525607ea570aa81c0a5bc68de09c8ea6eeb0e95f0R23)?

Reply:
discussed offline